lasagne/layers/recurrent.py

```diff
@@ -107,7 +107,11 @@ class CustomRecurrentLayer(MergeLayer):
         :class:`lasagne.layers.Layer` instance which connects input to the
         hidden state (:math:`f_i`).  This layer may be connected to a chain of
         layers, which must end in a :class:`lasagne.layers.InputLayer` with the
-        same input shape as `incoming`.
+        same input shape as incoming, except for the first dimension: When
+        ``precompute_input == True`` (the default), it must be
+        ``incoming.output_shape[0]*incoming.output_shape[1]`` or `None`; when
+        ``precompute_input == False``, it must be ``incoming.output_shape[0]``
+        or `None`.
```

Comment:
Ah sorry, didn't see you committed in between. This formulation is also fine with me! There are just two markup errors: "incoming" should have single backticks because it refers to the argument, and "None" should have double backticks because it's Python code.